### ISSUE TITLE
plugin was built with a different version of package golang.org/x/sys/unix.

### ISSUE THREAD

**Issue by markvdp** [gewv-tu-dresden/mosquitto-go-auth-oauth2#11]
Hi,

Build a container with  `docker` build --no-cache -t moasquitto-oauth2:latest` .

Output when starting the container with 'docker-compose up'
```
Attaching to mosquitto
mosquitto    | 1640690452: Loading config file /etc/mosquitto/conf.d/go-auth.conf
mosquitto    | 1640690452: mosquitto version 2.0.12 starting
mosquitto    | 1640690452: Config loaded from /etc/mosquitto/mosquitto.conf.
mosquitto    | 1640690452: Loading plugin: /mosquitto/go-auth.so
mosquitto    | 1640690452:  ├── Username/password checking enabled.
mosquitto    | 1640690452:  ├── TLS-PSK checking enabled.
mosquitto    | 1640690452:  └── Extended authentication not enabled.
mosquitto    | time="2021-12-28T11:20:52Z" level=warning msg="unknown or empty hasher, defaulting to PBKDF2"
mosquitto    | time="2021-12-28T11:20:52Z" level=fatal msg="Backend register error: couldn't initialize plugin backend with error could not init custom plugin: plugin.Open(\"/mosquitto/mosquitto-go-auth-oauth2\"): plugin was built with a different version of package golang.org/x/sys/unix."
mosquitto exited with code 1

```

I have no knowledge of 'go', Docker file looks oke, plugins are build with the same golang. What could be the cause of this? is it fixable? 

### Config

cat mosquitto.conf
``` 
# logging
log_type all

allow_anonymous false

# TCP
listener 1883

# Websocket
listener 9001
protocol websockets

include_dir /etc/mosquitto/conf.d

```

cat go-auth.conf
``` 
auth_plugin /mosquitto/go-auth.so

auth_opt_log_level debug
auth_opt_log_dest stdout
auth_opt_backends plugin
auth_opt_check_prefix false

auth_opt_password_path /etc/mosquitto/auth/passwords
auth_opt_acl_path /etc/mosquitto/auth/acls

auth_opt_cache_host redis
auth_opt_cache false
auth_opt_cache_reset true
# use redis DB 4 to avoid messing with other services.
auth_opt_cache_db 4

auth_opt_plugin_path /mosquitto/mosquitto-go-auth-oauth2.so
auth_opt_oauth_client_id rabbit_geranol
auth_opt_oauth_client_secret Fxxxxxxxx
auth_opt_oauth_token_url https://keycloak:8080/protocol/openid-connect/token
auth_opt_oauth_userinfo_url https://keycloak:8080/protocol/openid-connect/userinfo
auth_opt_oauth_cache_duration 600
# scopes can be a comma seperated list
auth_opt_oauth_scopes all

```

Small compose file
```
version: "3.3"

services:
  mosquitto:
    # image build with
    # docker build --no-cache -t mosquitto-oauth2 .
    image: mosquitto-oauth2:latest
    container_name: mosquitto
    command: "/usr/sbin/mosquitto -c /etc/mosquitto/mosquitto.conf"
    volumes:
      - ./mosquitto-config/:/etc/mosquitto

```

**Comment by Booyaabes** [comment]
Hi,

Same problem here, and I can't make it work either.

**Comment by Booyaabes** [comment]
OK, I managed to fix it. The problem is due to same dependencies in different versions for `mosquitto-go-auth` and `mosquitto-go-auth-oauth2`. 

To fix it, run the following in  `mosquitto-go-auth-oauth2` repository:

```sh
go get -v -u golang.org/x/sys
go get -v -u github.com/sirupsen/logrus
go mod tidy
```

In your Dockerfile, add the same commands before building `go-auth` plugin.

**Comment by karlTGA** [comment]
Hi, 

thank you for the report and sorry for the problems. I don't test the new version, so that goes on me.
@Booyaabes thank you for the hint. You brought me on track. The version of the logrus lib was in the `mosquitto-go-auth`  plugin `1.6.2` and i upgraded not knowingly to `1.8.0`. I set in the mod file the same version like in the mother plugin and on my system the produced container works. Please give me a feedback if you still have problems.

**Comment by Booyaabes** [comment]
Hi @karlTGA,

I can confirm that it works now. Thank you.

**Comment by karlTGA** [comment]
fixed by 84dd6c93f46466a2d7cd43e7622b4590b9701010
